warehouse/query-core/src/main/java/datawave/query/tables/keyword/StatefulKeywordUUIDChainStrategy.java

```diff
@@ -0,0 +1,104 @@
+import datawave.query.function.deserializer.DocumentDeserializer;
+import datawave.query.tables.chained.strategy.FullChainStrategy;
+
+public class StatefulKeywordUUIDChainStrategy extends FullChainStrategy<Entry<Key,Value>,Entry<Key,Value>> {
+    private int batchSize = -1;
+    protected DocumentDeserializer deserializer;
+
+    public StatefulKeywordUUIDChainStrategy(Query settings) {
+        this.deserializer = DocumentSerialization.getDocumentDeserializer(settings);
+    }
+
+    public int getBatchSize() {
+        return batchSize;
+    }
+
+    public void setBatchSize(int batchSize) {
+        this.batchSize = batchSize;
+    }
+
+    @Override
+    protected Query buildLatterQuery(Query initialQuery, Iterator<Entry<Key,Value>> initialQueryResults, String latterLogicName) {
+        log.debug("buildLatterQuery() called...");
+
+        String queryString = captureResultsAndBuildQuery(initialQueryResults, batchSize);
+
+        Query q = new QueryImpl(); // TODO, need to use a factory? don't hardcode this.
+        q.setQuery(queryString);
+        q.setId(UUID.randomUUID());
+        q.setPagesize(Integer.MAX_VALUE); // TODO: choose something reasonable.
+        q.setQueryAuthorizations(initialQuery.getQueryAuthorizations());
+        q.setUserDN(initialQuery.getUserDN());
+
+        q.addParameter(KeywordQueryLogic.CONTENT_VIEW_NAMES, initialQuery.findParameter(KeywordQueryLogic.CONTENT_VIEW_NAMES).getParameterValue());
+
+        return q;
+    }
+
+    public String captureResultsAndBuildQuery(Iterator<Entry<Key,Value>> initialQueryResults, int batchSize) {
+        int count = 0;
+        Set<String> queryTerms = new HashSet<>();
+        while (initialQueryResults.hasNext() && (batchSize == -1 || count < batchSize)) {
+            Entry<Key,Value> entry = initialQueryResults.next();
+            Entry<Key,Document> documentEntry = deserializer.apply(entry);
+
+            Key documentKey = documentEntry.getKey();
+
+            String row = documentKey.getRow().toString();
+            String colf = documentKey.getColumnFamily().toString();
+
+            int index = colf.indexOf("\0");
+            Preconditions.checkArgument(-1 != index);
+
+            String dataType = colf.substring(0, index);
+            String uid = colf.substring(index + 1);
+
+            Document document = documentEntry.getValue();
+            final Map<String,Attribute<? extends Comparable<?>>> documentData = document.getDictionary();
+
+            String identifier = null;
+            String language = null;
+
+            for (Entry<String,Attribute<? extends Comparable<?>>> data : documentData.entrySet()) {
+                if (data.getKey().equals("LANGUAGE")) {
+                    language = data.getValue().toString();
+                } else if (data.getKey().equals("HIT_TERM") && data.getValue() instanceof Content) {
+                    identifier = data.getValue().getData().toString();
+                }
+            }
```

Comment:
you may need a case to handle multi-valued fields here. In this case it might contain the target field rather than equaling it. I couldn't find an explicit example of this, but I feel like its out there.

Reply:
Because this class exists: https://github.com/NationalSecurityAgency/datawave/blob/integration/warehouse/query-core/src/main/java/datawave/query/attributes/Attributes.java

You get mess like this: https://github.com/NationalSecurityAgency/datawave/blob/integration/warehouse/query-core/src/main/java/datawave/query/attributes/Document.java#L309

Reply:
Done.